histogram/src/compact.rs

```diff
@@ -1,48 +1,362 @@
+                            }
+                            a = Some(serde::de::MapAccess::next_value::<u8>(&mut map)?);
+                        }
+                        Field::B => {
+                            if b.is_some() {
+                                return Err(<A::Error as serde::de::Error>::duplicate_field("b"));
+                            }
+                            b = Some(serde::de::MapAccess::next_value::<u8>(&mut map)?);
+                        }
+                        Field::N => {
+                            if n.is_some() {
+                                return Err(<A::Error as serde::de::Error>::duplicate_field("n"));
+                            }
+                            n = Some(serde::de::MapAccess::next_value::<u8>(&mut map)?);
+                        }
+                        Field::Index => {
+                            if index.is_some() {
+                                return Err(<A::Error as serde::de::Error>::duplicate_field(
+                                    "index",
+                                ));
+                            }
+                            index = Some(serde::de::MapAccess::next_value::<Vec<usize>>(&mut map)?);
+                        }
+                        Field::Count => {
+                            if count.is_some() {
+                                return Err(<A::Error as serde::de::Error>::duplicate_field(
+                                    "count",
+                                ));
+                            }
+                            count = Some(serde::de::MapAccess::next_value::<Vec<u64>>(&mut map)?);
+                        }
+                        _ => {
+                            let _ = serde::de::MapAccess::next_value::<serde::de::IgnoredAny>(
+                                &mut map,
+                            )?;
+                        }
+                    }
+                }
+
+                let a = a.ok_or_else(|| serde::de::Error::missing_field("a"))?;
+                let b = b.ok_or_else(|| serde::de::Error::missing_field("b"))?;
+                let n = n.ok_or_else(|| serde::de::Error::missing_field("n"))?;
+                let index = index.ok_or_else(|| serde::de::Error::missing_field("index"))?;
+                let count = count.ok_or_else(|| serde::de::Error::missing_field("count"))?;
+
+                if n > 64 {
+                    return Err(serde::de::Error::invalid_value(
+                        serde::de::Unexpected::Unsigned(n.into()),
+                        &"n must be in the range 0..=64",
+                    ));
+                }
+
+                if n < a + b {
+                    return Err(serde::de::Error::invalid_value(
+                        serde::de::Unexpected::Unsigned(n.into()),
+                        &"n must be greater than a + b",
+                    ));
+                }
+
```

Comment:
Shall we also check that index and count are the same length here, else the zip during conversion will presumably fail.

Reply:
The zip would succeed, but definitely need to check this anyway.

Fixed.